### PR TITLE
Add transport warmup example

### LINKED ISSUE #15: Provide example of how to do an inactive stream
S 4.1.4.1 reads:

"   Most web applications will not need to create answers using the
   "pranswer" type.  While it is good practice to send an immediate
   response to an "offer", in order to warm up the session transport and
   prevent media clipping, the preferred handling for a web application
   would be to create and send an "inactive" final answer immediately
   after receiving the offer.  Later, when the called user actually
   accepts the call, the application can create a new "sendrecv" offer
   to update the previous offer/answer pair and start the media flow.
   While this could also be done with an inactive "pranswer", followed
   by a sendrecv "answer", the initial "pranswer" leaves the offer-
   answer exchange open, which means the caller cannot send an updated
   offer during this time.
"

We need an example for this.

### PR DESCRIPTION
**Pull request by juberti**
Fixes #15

### CODE REVIEW

**Comment by juberti** [comment]
@fluffy, this is a reworked version of your original text. PTAL and let me know what you think.

File: draft-ietf-rtcweb-jsep.xml [hunk 1]
```diff
@@ -1275,38 +1275,41 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+            the caller cannot send an updated offer during this time.</t>
+
+            <t>As an example, consider a typical JSEP application that wants to
+            set up audio and video. When the callee
+            receives an offer with these MediaStreamTracks, it could send an
+            answer accepting the audio and video tracks as sendonly (meaning
+            that the caller will not send the callee any media yet, and because
+            the callee has not yet added its own MediaStreamTracks, the callee
+            will not send any media either). It could
+            then ask the user to accept the call, acquire the needed local
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: I feel like "could" -> "would" would make this graf stronger

File: draft-ietf-rtcweb-jsep.xml [hunk 2]
```diff
@@ -4444,6 +4447,152 @@ a=msid:-
...
       </section>
     </section>
+    <section title="Early Transport Warmup Example" anchor="sec.warmup-example" >
+
+      <t>This example demonstrates the early warmup technique described in
+      <xref target="sec.use-of-provisional-answer"/>. Here, Alice's endpoint
+      sends an offer to Bob's endpoint to start an audio/video call. Bob
+      immediately responds with an answer that accepts the audio/video
+      m= sections, but marks them as sendonly (from its perspective), meaning
+      that Alice will not yet send media. This allows the browser to start
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: s/browser/JSEP implementation/

File: draft-ietf-rtcweb-jsep.xml [hunk 3]
```diff
@@ -4444,6 +4447,152 @@ a=msid:-
...
+      negotiating ICE and DTLS immediately. Bob's endpoint then prompts him
+      to answer the call, and when he does, his endpoint sends a second offer
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: s/endpoint/JSEP implementation/?

File: draft-ietf-rtcweb-jsep.xml [hunk 4]
```diff
@@ -4444,6 +4447,152 @@ a=msid:-
...
+
+//                  ICE and DTLS establish while call is ringing
+
+//                  Bob accepts call, starts media, and sends a new offer
+BobJS->BobUA:       transceiver.setTrack with audio and video tracks
+BobJS->BobUA:       transceiver.setDirection(sendrecv) for both transceivers
+BobJS->BobUA:       createOffer
+BobJS->BobUA:       setLocalDescription with offer
+BobUA->AliceUA:     media sent from Bob to Alice
+[TODO: ice restart]
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Note this TODO. I'm assuming that's intentional

Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: yes - trying to avoid a mammoth PR

File: draft-ietf-rtcweb-jsep.xml (hunk 2 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: fixed

File: draft-ietf-rtcweb-jsep.xml (hunk 3 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: I think endpoint is probably most appropriate here - only applications/endpoints can trigger the sending of an offer.

File: draft-ietf-rtcweb-jsep.xml (hunk 1 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: The text is treating the approach as hypothetical, i.e. the app *could* do this if one wanted to.

Alternatively I could make it concrete and say something like 'if the app wants to do warmup, it does the following..."

Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: I reworded since Cullen had comments in this area as well.

**Comment by juberti** [comment]
Cullen PTAL

**Pull request merged**